### DISCUSSION TITLE
Option to pin a comment

### DISCUSSION THREAD

**Discussion by Andre601** [giscus/giscus#243] [Ideas]
It could be a beneficial thing to have a way of pinning a comment, so that it always appears at the top of the history, no matter what the possible sorting (if there is any) would be.

To not require any database storing or similar would I have two possible ideas to share:

### HTML Comment tag
The Bot could use a HTML comment tag, to "mark" a comment as a pinned one by editing the user's content and f.e. add `<!--giscus:pinned-->` to it, so that when it reads it, it finds the tag and makes the comment pinned.

There are however a few downsides:

- Everyone could post a comment with that tag. A solution would be to sanitize the comment by removing any comment tags they may post or edit in.
- This would add extra edits to the comment which would need to either be filtered out or just be shown.
- Changing the pinned comment would either require an unpin and then pinning, or require the bot to find the old comment, remove the tag and then add it to the newly pinned one. Tl;dr more work for the bot.

### Using "Mark as Answered"
This isn't a good idea for me, but I think it could still have benefits.
The bot can mark a comment as answer which it would treat as a pinning.

Downsides:

- Would require the category to be open, allowing anyone to just post their own discussions in them.
- People who post their own discussion (and perhaps manage to match a post's title) could just mark their own comment as an answer to pin it.

**Comment by JakeSteam** [comment]
What about the bot adds the "pinned" comment's ID (or URL, etc) into the original message, then looks for it? This would be vaguely similar to the [existing `data-strict` setup](https://github.com/giscus/giscus/blob/main/ADVANCED-USAGE.md#data-strict), and avoids all of the downsides you mentioned. 

Alternatively, it could be triggered & maintained by something simple like "if a repo admin has reacted 🚀 to any comment, treat the first one as pinned". This would also avoid all downsides mentioned, and likely be a bit simpler.

**Reply by Andre601** [reply]
When I understand the first one right do you mean that the bot would keep a HTML comment or smth with the data (i.e. comment id) in its initial post? That could actually be a decent solution to have...

**Reply by JakeSteam** [reply]
Exactly, it could use (in a hidden bit of the message) the ID of the comment to "pin". The challenge would be how to set that, which perhaps could use the emoji-react bit I mentioned. 

If the system uses emoji reactions though, no need for any hidden IDs!

**Reply by laymonage** [reply]
There is currently no way to get the pinned comment without scanning through the whole discussion. If you have a lot of comments (e.g. >100), some comments will be hidden in pagination, one of which may be the pinned comment. [The API](https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions) doesn't let you get a specific comment by ID.

**Comment by vineet-k09** [comment]
Its totally upto them guys
